apps/crn-frontend/src/network/teams/TeamOutput.tsx

```diff
@@ -56,28 +62,57 @@ export function paramOutputDocumentTypeToResearchOutputDocumentType(
 
 type TeamOutputProps = {
   teamId: string;
+  researchOutputData?: ResearchOutputResponse;
 };
-const TeamOutput: React.FC<TeamOutputProps> = ({ teamId }) => {
+const TeamOutput: React.FC<TeamOutputProps> = ({
+  teamId,
+  researchOutputData,
+}) => {
   const paramOutputDocumentType = useParamOutputDocumentType(teamId);
-  const documentType = paramOutputDocumentTypeToResearchOutputDocumentType(
-    paramOutputDocumentType,
-  );
+  const documentType =
+    researchOutputData?.documentType ||
+    paramOutputDocumentTypeToResearchOutputDocumentType(
+      paramOutputDocumentType,
+    );
   const team = useTeamById(teamId);
   const [errors, setErrors] = useState<ValidationErrorResponse['data']>([]);
 
   const { canCreateUpdate } = useContext(ResearchOutputPermissionsContext);
 
   const createResearchOutput = usePostTeamResearchOutput();
+  const updateResearchOutput = usePutTeamResearchOutput(
+    researchOutputData?.id as string,
```

Comment:
I don't like this cast. Could we not use the ID from the URL which is guaranteed to be there?